packages/parse5/lib/parser/open-element-stack.test.ts

```diff
@@ -317,6 +317,8 @@ generateTestsForEachTreeAdapter('open-element-stack', (treeAdapter) => {
     test('Has numbered header in scope', () => {
         const stack = new OpenElementStack(treeAdapter.createDocument(), treeAdapter, stackHandler);
 
+        assert.ok(stack.hasNumberedHeaderInScope());
```

Comment:
The empty states won't be used in the parser, but we can test them here.

Reply:
🤔 Why does `hasNumberedHeaderInScope` exist if parse5 doesn’t use it? Should we deprecate it and remove it in a major then?

Reply:
If these functions are used, can we instead add unit tests (input/output) of parsing behavior that depends on the result of these functions?

Reply:
`hasNumberedHeaderInScope` is used by the parser. The only thing that the parser cannot reach is the default return value, eg. for an empty stack.

Reply:
You mean this early return https://github.com/inikulin/parse5/blob/8e22fe40d5774e2f2d5eb9be230f4f78d60e780d/packages/parse5/lib/parser/open-element-stack.ts#L270 causing https://github.com/inikulin/parse5/blob/8e22fe40d5774e2f2d5eb9be230f4f78d60e780d/packages/parse5/lib/parser/open-element-stack.ts#L278 to never run?

Could you change that early return to a break in that case, to fix the coverage?

Not sure about the algorithm: but it seems off that the default for that function i `true`? I’d expect, with such a function  name, to exit early with `true`, and otherwise return `false`?

Reply:
The relevant part of the spec is https://html.spec.whatwg.org/multipage/parsing.html#has-an-element-in-scope

The algorithm explicitly doesn't consider the possibility of an empty stack:

> This will never fail, since the loop will always terminate in the previous step if the top of the stack — an [html](https://html.spec.whatwg.org/multipage/semantics.html#the-html-element) element — is reached.

Reply:
I have updated the logic to use `break`, but am still split if this is the right approach. Another option would be to error if the loop terminates.

Reply:
Then I recommend using a `c8 ignore` (or similar), with this as an explanation:

```
/* Note: the algorithm <link> never fails */
/* c8 ignore next 123 */
```